storage/gcloud/aio/storage/storage.py

```diff
@@ -172,6 +173,22 @@ async def _headers(self) -> Dict[str, str]:
             'Authorization': f'Bearer {token}',
         }
 
+    async def get_buckets(
+        self, project: str, timeout: int = DEFAULT_TIMEOUT,
+    ) -> Set[Bucket]:
+        url = ('https://storage.googleapis.com/'
+               f'storage/v1/b?project={project}')
+
+        headers = await self._headers()
+        resp = await self.session.get(url,
+                                      timeout=timeout, headers=headers)
+
+        data = await resp.json(content_type=None)
+        result = set()
+        for item in data['items']:
+            result.add(Bucket(self, item['id']))
+        return result
```

Comment:
The [docs](https://cloud.google.com/storage/docs/json_api/v1/buckets/list#parameters) mention that this AIP is paginated and that the `nextPageToken` value must be checked to get all results.

You should be able to steal the code from our `list_blobs` method [here](https://github.com/talkiq/gcloud-aio/blob/0dc6f57ac6af6f5859c9feeec3d75703ddf39572/storage/gcloud/aio/storage/bucket.py#L64-L82).

Reply:
Thanks for the example. I changed it accordingly.